kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/TaskInstance.java

```diff
@@ -64,6 +64,8 @@
     private Long parentId;
     @XmlElement(name="task-process-id")
     private String processId;
+    @XmlElement(name="process-instance-desc")
+    private String processInstanceDescription;
```

Comment:
I don't like this one... it essentially means always a join to have this information which does not make much sense in my opinion. In other words process instance description does not belong to task instance so having it on TaskInstance type is misleading.

Reply:
One of the custom queries register for the customer using the processInstanceDescription as a businessId and it's needed to find all the task related to this businesId (which is shared between several processes related to the same contract). If I don't add the processInstanceDescription to the TaskInstance model, how can we get the processInstanceDescription using the mapper I implemented and to which I added it?  (https://github.com/kiegroup/jbpm/pull/1146)

Reply:
as described, you either create new type (TaskWithProcessInstance most likely better name would need to be used) and then map it here when using your additional jbpm service mappers. Or use the Raw mapper that will return array of arrays. Raw mapper is fine when KIE Server client is not used.

Reply:
@mswiderski  I added a new type with the custom data used in my addition jbpm service mapper. Wdyt?